packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -120,8 +121,13 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
 
         // Measure, and update to show the items until maxRows is reached.
         yield computeVisibleTagCount();
+
+        // Set a fixed height on the container
+        setMaxHeight(containerRef.current?.getBoundingClientRect().height);
```

Comment:
So to reproduce the original issue:
- In the maxRows story, grow the container, then shrink it again. As it shrinks, there is one render where the button will go to the next line (creating a flicker and potentially causing a layout shift)

The intent of this fix is to add a static height to the container (and overflow:hidden), so that no flickering or layout shifts will occur for that extra render while resizing. 

We could probably keep that static height value in a ref to avoid extra renders.

Ideally we would want to avoid that extra render in the first place, but I'm not sure how we would do that (open to ideas.)

Reply:
I think we could take the height+margins of a single tag and multiply it by the maxRows to determine this value earlier. It could be included in the returned yielded state.

It stops the flickering on a new line, but it hasn't quite stopped the flickering on the current line.
If we kept an identical copy of the tags in a visually hidden div, we could do our measurements on that, then just apply the result to the visible one. That would make it appear as though it just arrived at the correct spot immediately. We might still want/need the maxHeight.